chain/client/src/client.rs

```diff
@@ -427,7 +429,10 @@ impl Client {
         block: &Block,
     ) -> Result<(), Error> {
         let epoch_id = self.epoch_manager.get_epoch_id(block.hash())?;
-        let shard_layout = self.epoch_manager.get_shard_layout(&epoch_id)?;
+        let protocol_version = self.epoch_manager.get_epoch_protocol_version(&epoch_id)?;
+        let shard_layout =
+            self.epoch_manager.get_shard_layout_from_protocol_version(protocol_version);
```

Comment:
ditto

Reply:
I need to look up the runtime config below for which I need the protocol version.  Once I have the protocol version, I can reduce one `?` which I hate =).